src/test/java/com/aws/greengrass/security/SecurityServiceTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.security;
+
+import com.aws.greengrass.config.CaseInsensitiveString;
+import com.aws.greengrass.security.exceptions.ServiceProviderConflictException;
+import com.aws.greengrass.security.exceptions.ServiceUnavailableException;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import org.hamcrest.collection.IsMapContaining;
+import org.hamcrest.collection.IsMapWithSize;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import javax.net.ssl.KeyManager;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({MockitoExtension.class, GGExtension.class})
+class SecurityServiceTest {
+
+    private final SecurityService service = new SecurityService();
+
+    @Mock
+    private CryptoKeySpi mockKeyProvider;
+
+    @Test
+    void GIVEN_key_service_provider_WHEN_register_provider_THEN_succeed() throws Exception {
+        when(mockKeyProvider.supportedKeyType()).thenReturn("FILE");
+        service.registerCryptoKeyProvider(mockKeyProvider);
+        assertThat(service.getCryptoKeyProviderMap(), IsMapWithSize.aMapWithSize(1));
+        assertThat(service.getCryptoKeyProviderMap(), IsMapContaining.hasEntry(new CaseInsensitiveString("file"),
+                mockKeyProvider));
```

Comment:
Add a new key type as well to assert that 2nd key type gets added successfully?

Reply:
will do